FindWithScalarFields/src/FindWithScalarFields.java

```diff
@@ -0,0 +1,207 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Projections;
+
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.gt;
+import static com.mongodb.client.model.Filters.lt;
+import static com.mongodb.client.model.Filters.gte;
+import static com.mongodb.client.model.Filters.lte;
+import static com.mongodb.client.model.Projections.fields;
+import static com.mongodb.client.model.Projections.include;
+import static com.mongodb.client.model.Projections.excludeId;
+
+public class FindWithScalarFields {
+	
+	/**
+	 * Retrieves documents filtered by shipping address postal.
+	 * @param connectionString: To MongoDB instance/MongoDB Cluster.
+	 * @param postalcode: Shipping address postal code to filter documents.
+	 * @return List<Document>: ArrayList of matching documents.
+	 */
+	public List<Document> GetOrdersByPostalCode(String connectionString, int postalcode){
+		if(connectionString == null || connectionString.isEmpty() || postalcode<0)
+		{
+			throw new IllegalArgumentException();
+		}
+		else
+		{
+			MongoClientURI clientUri = new MongoClientURI(connectionString);
+			try(MongoClient client = new MongoClient(clientUri))
+			{
+				MongoDatabase database = client.getDatabase("stores");			
+				MongoCollection<Document> collection = database.getCollection("orders");		
+				
+				//option 1 - Static import the required filters
+				List<Document> ordersFiltered = collection
+						.find(eq("shippingAddress.postalCode",postalcode))
+						.into(new ArrayList<Document>());
+				
+				//option 2  - create bson filter
+				/*Bson filter = new Document("shippingAddress.postalcode",postalcode);
+				List<Document> ordersFiltered = collection
+						.find(filter)
+						.into(new ArrayList<Document>());*/
+				
+				return ordersFiltered;
+			}
+			catch (Exception e) {
+				System.out.println("Exception occured");
+				System.out.println("Deatils:");
+				System.out.println(e.getStackTrace());
+				return null;
+			}	
+		}			
+	}
+	
+	/**
+	 * Retrieves documents filtered by shipping address postal code and total.
+	 * @param connectionString: To MongoDB instance/MongoDB Cluster.
+	 * @param postalcode: Shipping address postal code to filter documents.
+	 * @param total: Order total to filter documents.
+	 * @return List<Document>: ArrayList of matching documents.
+	 */
+	public List<Document> GetOrdersByPostalCodeAndTotal(String connectionString, int postalcode, Double total){
```

Comment:
Why did you choose this particular query? Do you believe it is common that people who use our system would frequently want to search both those those fields.

Reply:
Also searching by total except in a range query would be very uncommon.

Reply:
Need different examples here. Need examples that are easily to complete, but provide readers with a plausible scenario.